google-ads/src/main/java/com/google/ads/googleads/lib/utils/FieldMasks.java

```diff
@@ -76,12 +75,14 @@ private static void compare(
           mask.addPaths(fieldName);
         }
       } else {
+        boolean hasValueChanged =
+            original.hasField(field) != modified.hasField(field)
```

Comment:
Do the calls to `original.getField(field)` and `modified.getField(field)` above change the presence bit? I see those return the default value if the field is not set, but I don't know if they have any side effects.

Reply:
Good question, but no since protobuf messages in Java are immutable.

Reply:
Thanks, this change LGTM. Going forward we may need to revisit since the protobuf team recently was discouraging the use of `hasField()` checks, but I only expect that to become a concern when and if we remove the `optional` keyword on mutable fields.